src/DDTrace/OpenTelemetry/SpanBuilder.php

```diff
@@ -92,6 +96,38 @@ public function addLink(SpanContextInterface $context, iterable $attributes = []
         return $this;
     }
 
+    public function addEvent(string $name, int $timestamp = null, iterable $attributes = []): SpanBuilderInterface
+    {
+        $attributesArray = Attributes::create($attributes);
+        $nanoTimestamp = ($timestamp !== null && $timestamp < 1e9) ? $timestamp * 1000 : ($timestamp ?? (int)(microtime(true) * 1e9)); // Convert microseconds to nanoseconds if needed
```

Comment:
That's a weird condition. What is this `$timestamp < 1e9` check for? As far as I see times in OTel are generally nanoseconds.